dipy/viz/app.py

```diff
@@ -0,0 +1,519 @@
+        True,
+        "//VTK::Coincident::Dec\n"
+        "uniform float selected;\n"
+        "uniform float opacity_level;\n",
+        False)
+
+    @window.vtk.calldata_type(window.vtk.VTK_OBJECT)
+    def vtk_shader_callback(caller, event, calldata=None):
+        program = calldata
+        if program is not None:
+            try:
+                program.SetUniformf("selected",
+                                    hz.cea[actor]['selected'])
+            except KeyError:
+                pass
+            try:
+                program.SetUniformf("selected",
+                                    hz.cla[actor]['selected'])
+            except KeyError:
+                pass
+            program.SetUniformf("opacity_level", 1)
+
+    gl_mapper.AddObserver(window.vtk.vtkCommand.UpdateShaderEvent,
+                          vtk_shader_callback)
+
+
+HELP_MESSAGE = """
+>> left click: select centroid
+>> e: expand centroids
+>> r: collapse open clusters
+>> h: hide unselected centroids
+>> i: invert selection
+>> a: select all centroids
+>> s: save in file
+"""
+
+
+class Horizon(object):
+
+    def __init__(self, tractograms, images, cluster, cluster_thr,
+                 random_colors, length_lt, length_gt, clusters_lt, clusters_gt,
+                 world_coords=True, interactive=True, out_png='tmp.png'):
+        """ Highly interactive visualization - invert the Horizon!
+
+        Parameters
+        ----------
+        tractograms : sequence
+            Sequence of Streamlines objects
+        images : sequence of tuples
+            Each tuple contains data and affine
+        cluster : bool
+            Enable QuickBundlesX clustering
+        cluster_thr : float
+            Distance threshold used for clustering
+        random_colors : bool
+        length_lt : float
+        length_gt : float
+        clusters_lt : int
+        clusters_gt : int
+        world_coords : bool
+        interactive : bool
+        out_png : string
+
+        References
+        ----------
+        .. [Horizon_ISMRM19] Garyfallidis E., M-A. Cote, B.Q. Chandio,
+            S. Fadnavis, J. Guaje, R. Aggarwal, E. St-Onge, K.S. Juneja,
+            S. Koudoro, D. Reagan, DIPY Horizon: fast, modular, unified and
+            adaptive visualization, Proceedings of: International Society of
+            Magnetic Resonance in Medicine (ISMRM), Montreal, Canada, 2019.
+        """
+
+        self.cluster = cluster
+        self.cluster_thr = cluster_thr
+        self.random_colors = random_colors
+        self.length_lt = length_lt
+        self.length_gt = length_gt
+        self.clusters_lt = clusters_lt
+        self.clusters_gt = clusters_gt
+        self.world_coords = world_coords
+        self.interactive = interactive
+        self.prng = np.random.RandomState(27)
+        self.tractograms = tractograms
+        self.out_png = out_png
+
+        if images is None:
+            self.images = []
+        else:
+            self.images = images
```

Comment:
I think you can use the pythonic way `self.images = images or []` instead of `if/else` directive